### ISSUE TITLE
Cacti Aggregate Graph Problem When Added Under a Tree

### ISSUE THREAD

**Issue by takobaba** [Cacti/cacti#1821]
Hi Guys,

I have a really weird problem. I have created an aggregate graph from simple 2 line graphs. When the aggregate graph is created under the aggregate graphs section(under management) everything is cool and working accordingly. 

However, when I add the aggregated graph under a graph tree, the whole RRD for the aggregated graph changes and the graph cannot show the total value.

Here is the graph rrd before adding to a tree:

Code:
```
/bin/rrdtool graph - \
--imgformat=PNG \
--start='-86400' \
--end='-300' \
--title='Total Online Users' \
--base='1000' \
--height='150' \
--width='650' \
--alt-autoscale-max \
--lower-limit='0' \
--units-exponent='0' \
--vertical-label='Online Users' \
--slope-mode \
--font TITLE:10: \
--font AXIS:7: \
--font LEGEND:8: \
--font UNIT:7: \
DEF:a='/usr/share/cacti/rra/xxxxxxxxxxxxxxxx.rrd':'dsl_user_input':AVERAGE \
DEF:b='/usr/share/cacti/rra/yyyyyyyyyyyyyyyy.rrd':'time':AVERAGE \
CDEF:cdefa='a,0,*' \
CDEF:cdefb='b,0,*' \
CDEF:cdefc='TIME,1533625072,GT,a,a,UN,0,a,IF,IF,TIME,1533625072,GT,b,b,UN,0,b,IF,IF,+' \
LINE2:cdefa#0000FFFF:''  \
LINE2:cdefb#0000FFFF:''  \
GPRINT:cdefc:LAST:'Total Users\:%8.0lf'  \
LINE2:cdefc#0000FFFF:''  \
GPRINT:cdefc:MIN:'Min Users\: %8.0lf'  \
GPRINT:cdefc:MAX:'Max Users\: %8.0lf\n'
```

Here is the graph rrd after adding to a tree:
Code:
```
/bin/rrdtool graph - \
--imgformat=PNG \
--start='-86400' \
--end='-300' \
--title='Total Online Users' \
--base='1000' \
--height='150' \
--width='650' \
--alt-autoscale-max \
--lower-limit='0' \
--units-exponent='0' \
--vertical-label='Online Users' \
--slope-mode \
--font TITLE:10: \
--font AXIS:7: \
--font LEGEND:8: \
--font UNIT:7: \
DEF:a='/usr/share/cacti/rra/xxxxxxxxxxxxxxxx.rrd':'dsl_user_input':AVERAGE \
DEF:b='/usr/share/cacti/rra/yyyyyyyyyyyyyyyy.rrd':'time':AVERAGE \
CDEF:cdefa='a,0,*' \
CDEF:cdefb='b,0,*' \
LINE2:cdefa#0000FFFF:''  \
LINE2:cdefb#0000FFFF:''
```

Basically the cdef function loses the total value(cdefc) and the graph ends up showing nothing. 

I can't understand why!

Let me know if any further information is needed.

Any help appreciated!
Thanks!!!

**Comment by cigamit** [comment]
Well, the preview uses lib/rrd.php just like everything else.  Is the Aggregate template based or a standalone one?

**Comment by takobaba** [comment]
I have tried both standalone and template, result is the same.

**Comment by takobaba** [comment]
what you are asking is already in the first post. before and after codes showing the rrd functions after enabling the debug.

before code is the debug from the preview page
after code is the debug from the graph view

I agree, I cant understand either. why would the rrd changes based on the view. maybe it uses a different rrd.php? I have searched there is only one rrd.php file.

I am clueless.....

**Comment by netniV** [comment]
Hi @takobaba, additionally if they are definitely the same graph, is there any chance that you could send a dump of your database?  That way we can properly investigate the difference as it seems to lie with your specific data.

**Comment by cigamit** [comment]
No response.
